src/ucs/sys/sys.c

```diff
@@ -881,7 +881,12 @@ ucs_status_t ucs_sysv_alloc(size_t *size, size_t max_size, void **address_p,
 
     /* Attach segment */
     if (*address_p) {
+#ifdef SHM_REMAP
         ptr = shmat(*shmid, *address_p, SHM_REMAP);
+#else
```

Comment:
I don't think we can just fail here. Non null pointer can be passed from upper level https://github.com/openucx/ucx/blob/e23e2122d07a818ff2bc3db68c4f0e7f43b7839a/src/uct/base/uct_mem.c#L223

I suspect that this flow is used by SHMEM that aims to manage the same address across all PEs

Reply:
which why we have to fail here, cannot use sysv method to allocate hepa in this case, must fallback to other method (eg mmap)

Reply:
Maybe just disable sysv all togher so you don't have to deal with some allocation in sysv and another in mmap ?

Reply:
IMO no reason to disable completely. MPI shared memory can still use it.

Reply:
ok.